### PR TITLE
Add properties to user resource for certificate creation

### PR DESCRIPTION
**Pull request by daviddjh**
# Description

This change adds properties to the user resource of the openvpn cookbook. As a result, developers can specifically define attributes of the user certificate they want to create within a resource statement. Environment variables are updated in the server recipe to provide default values.

### CODE REVIEW

File: test/integration/server_verification/default_test.rb [hunk 1]
```diff
@@ -10,3 +10,16 @@
...
+      subject { super().content }
+      it { is_expected.to include 'C=CA' }
+      it { is_expected.to include 'ST=Ontario' }
+      it { is_expected.to include 'L=Ottawa' }
+      it { is_expected.to include 'O=Test Org' }
+      it { is_expected.to include 'OU=Test Org Unit' }
+      it { is_expected.to include 'CN=vpn_user/emailAddress=vpn_user@test.com' }
+      it { is_expected.to include 'Public-Key: (1024 bit)' }
+    end
+end
```
Comment (detjensrobert) [inline test/integration/server_verification/default_test.rb:25]: These should be
```rb
its('content') { should match /stuff/ }
```
to match InSpec convention.

File: recipes/server.rb [hunk 2]
```diff
@@ -99,7 +99,17 @@
 end
 
 bash 'openvpn-initca' do
-  environment('KEY_CN' => "#{node['openvpn']['key']['org']} CA")
```
Comment (detjensrobert) [inline recipes/server.rb:102]: Any reason why some of these are `bash` resources and others are `execute`? Might be good to switch them all to `execute` for consistency.

**Review by detjensrobert — CHANGES_REQUESTED** [review]

File: templates/openssl.cnf.erb [hunk 3]
```diff
@@ -9,7 +9,7 @@ engines                 = engine_section
 [ ca ]
 default_ca	= CA_default
 [ CA_default ]
-dir		= <%= node['openvpn']['key_dir'] %>
+dir		= $ENV::KEY_DIR
```
Comment (Xorima) [inline templates/openssl.cnf.erb:12]: This feels wrong, we should be just accessing the vars directly, not using envs?

Comment (daviddjh) [reply templates/openssl.cnf.erb:12]: Setting the vars directly during static file generation seems appropriate if you'd like a cleaner, more universal openssl.cnf. However, as far as I can tell, this limits the configurability of user certificates. Previously, this resource was using envs to set the Common Name. Now, more envs are used so the resource can set more key information per call, although I could understand not needing this for key_dir. The use of envs specifically is inspired by pkitool's openssl cnf linked [here](https://github.com/OpenVPN/easy-rsa-old/blob/master/easy-rsa/2.0/openssl-1.0.0.cnf). The cookbook attributes are still used for configuration defaults, however, anyone using the user resource can now override them. Please let me know if adding envs to allow for more customization of user certificates isn't appropriate here.

File: resources/user.rb [hunk 4]
```diff
@@ -8,18 +8,27 @@
 property :force, [true, false]
 property :destination, String
 property :additional_vars, Hash, default: {}
+property :key_dir,   String, default: lazy { node['openvpn']['key_dir'] }
```
[the remark this answers is not in the document]
Comment (daviddjh) [reply resources/user.rb:11]: Good idea, I'll try it out

File: recipes/server.rb (hunk 2 above)
Comment (daviddjh) [reply recipes/server.rb:102]: Not as far as I can tell, I'll change that.

File: templates/openssl.cnf.erb (hunk 3 above)
Comment (Xorima) [reply templates/openssl.cnf.erb:12]: I think it is best to ask @sous-chefs/board to weigh in here

Comment (ramereth) [reply templates/openssl.cnf.erb:12]: I think it's OK to use vars here, maybe not for this one specifically but the others are OK IMO.

File: resources/user.rb [hunk 5]
```diff
@@ -27,17 +27,18 @@
...
-      'KEY_SIZE' => node['openvpn']['key']['size'].to_s,
-      'KEY_COUNTRY' => node['openvpn']['key']['country'],
-      'KEY_PROVINCE' => node['openvpn']['key']['province'],
-      'KEY_CITY' => node['openvpn']['key']['city'],
-      'KEY_ORG' => node['openvpn']['key']['org'],
-      'KEY_EMAIL' => node['openvpn']['key']['email']
+      'EASY_RSA'     => '/etc/openvpn/easy-rsa',
+      'KEY_CONFIG'   => '/etc/openvpn/easy-rsa/openssl.cnf',
+      'KEY_DIR'      => key_dir,
+      'CA_EXPIRE'    => (new_resource.key_vars['ca_expire'] ? new_resource.key_vars['ca_expire'].to_s : node['openvpn']['key']['ca_expire'].to_s),
```
Comment (detjensrobert) [inline resources/user.rb:33]: These ternaries can be simplified to a OR:
```suggestion
      'CA_EXPIRE'    => (new_resource.key_vars['ca_expire'] || node['openvpn']['key']['ca_expire']).to_s,
```

**Review by detjensrobert — CHANGES_REQUESTED** [review]

**Review by detjensrobert — APPROVED** [review]
Looks good, pending on ci updates from #200

**Review by ramereth — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [6.1.0](https://github.com/sous-chefs/openvpn/releases/tag/6.1.0)

**Pull request merged**